src/main/java/com/jfrog/ide/common/scan/GraphScanLogic.java

```diff
@@ -98,25 +100,35 @@ DependencyTree createScanTree(DependencyTree root) {
     private void populateScanTree(DependencyTree root, DependencyTree scanTree, Set<String> componentsAdded) {
         for (DependencyTree child : root.getChildren()) {
             // Don't add metadata nodes to the scan tree
-            if (child.isMetadata()) {
-                populateScanTree(child, scanTree, componentsAdded);
-                continue;
-            }
-
-            // Add the dependency subtree to the scan tree
-            String childFullId = child.toString();
-            if (((DependencyTree) child.getParent()).isMetadata()) {
-                // All direct dependencies should be in the cache. This line make sure that dependencies that
-                // wouldn't return from Xray will not be scanned again during the next quick scan.
-                String componentId = contains(childFullId, "://") ?
-                        substringAfter(childFullId, "://") : childFullId;
-            }
-            if (componentsAdded.add(child.getComponentId())) {
-                scanTree.add(new DependencyTree(child.getComponentId()));
+            if (!child.isMetadata()) {
+                // Add the dependency subtree to the scan tree
+                if (componentsAdded.add(child.getComponentId())) {
+                    scanTree.add(new DependencyTree(child.getComponentId()));
+                }
             }
             populateScanTree(child, scanTree, componentsAdded);
```

Comment:
Should we populateScanTree if it isn't the first time we "see" this dependency?
`(componentsAdded.add(child.getComponentId() == false)`